### PR TITLE
Update Faction Leaders Abilities

### PR DESCRIPTION
**Pull request by NickTyrer**
This PR syncs faction leaders with https://github.com/cmangos/tbc-db/commit/c5a6c3f9d5cf77b7a1e798180282567be7a697c5

### CODE REVIEW

File: sql/migrations/20201010150010_world.sql [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+(128401, 0, 16, 5883, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Archbishop Benedictus - Play Sound');
+INSERT INTO `creature_ai_events` (`id`, `creature_id`, `condition_id`, `event_type`, `event_inverse_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action1_script`, `action2_script`, `action3_script`, `comment`) VALUES (128401, 1284, 0, 4, 0, 100, 0, 0, 0, 0, 0, 128401, 0, 0, 'Archbishop Benedictus - Play Sound 5883 on Aggro');
+
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(128403, 0, 47, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Archbishop Benedictus - Flee'),
+(128403, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -47, 0, 0, 0, 0, 0, 0, 0, 0, 'Archbishop Benedictus - Say Text');
```
Comment (Daribon) [inline sql/migrations/20201010150010_world.sql]: Should be broadcast text id 1150 instead of -47.

File: sql/migrations/20201010150010_world.sql [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(496806, 0, 15, 20692, 8, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Lady Jaina Proudmoore - Cast Spell');
+INSERT INTO `creature_ai_events` (`id`, `creature_id`, `condition_id`, `event_type`, `event_inverse_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action1_script`, `action2_script`, `action3_script`, `comment`) VALUES (496806, 4968, 0, 9, 0, 100, 1, 0, 50, 2000, 3500, 496806, 0, 0, 'Lady Jaina Proudmoore - Cast Fireball');
+UPDATE `creature_template` SET `ai_name` = 'EventAI' WHERE `entry` = 4968;
+
+
+-- Vol'jin 10540
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: He already uses all these abilities from his spell list.

File: sql/migrations/20201010150010_world.sql [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(128406, 0, 15, 20695, 8, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Archbishop Benedictus - Cast Spell');
+INSERT INTO `creature_ai_events` (`id`, `creature_id`, `condition_id`, `event_type`, `event_inverse_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action1_script`, `action2_script`, `action3_script`, `comment`) VALUES (128406, 1284, 0, 9, 0, 100, 1, 0, 40, 3400, 4800, 128406, 0, 0, 'Archbishop Benedictus - Cast Holy Smite');
+UPDATE `creature_template` SET `ai_name` = 'EventAI' WHERE `entry` = 1284;
+
+
+-- Lady Jaina Proudmoore 4968
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: Jaina has a c++ script and uses all these abilities from there. If you want to move it to DB, that's okay, but simple timed casts belong in a spell list, not in Event AI. Do not blindly copy scripts from cmangos. You even left the comments the same, even though we have no such thing as "Ranged Mode". We have a flag on spells inside spell lists, that makes creatures stop moving and melee attacking until the spell cast fails, which produces the same behavior in game. Also if you move her script to db, you will need to hack spell 20682 to remove teleported players from combat, or Jaina will try to chase them all the way to the docks which will cause her to reset.

File: sql/migrations/20201010150010_world.sql [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+UPDATE `creature_template` SET `spell_list_id` = 79370 WHERE `entry` = 35160;
+
+
+-- Archbishop Benedictus 1284
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: These queries throw an error, as he already has an Event AI script.

File: sql/migrations/20201010150010_world.sql [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
...`delayInitialMax_4`, `delayRepeatMin_4`, `delayRepeatMax_4`, `scriptId_4`, `spellId_5`, `probability_5`, `castTarget_5`, `targetParam1_5`, `targetParam2_5`, `castFlags_5`, `delayInitialMin_5`, `delayInitialMax_5`, `delayRepeatMin_5`, `delayRepeatMax_5`, `scriptId_5`, `spellId_6`, `probability_6`, `castTarget_6`, `targetParam1_6`, `targetParam2_6`, `castFlags_6`, `delayInitialMin_6`, `delayInitialMax_6`, `delayRepeatMin_6`, `delayRepeatMax_6`, `scriptId_6`, `spellId_7`, `probability_7`, `castTarget_7`, `targetParam1_7`, `targetParam2_7`, `castFlags_7`, `delayInitialMin_7`, `delayInitialMax_7`, `delayRepeatMin_7`, `delayRepeatMax_7`, `scriptId_7`, `spellId_8`, `probability_8`, `castTarget_8`, `targetParam1_8`, `targetParam2_8`, `castFlags_8`, `delayInitialMin_8`, `delayInitialMax_8`, `delayRepeatMin_8`, `delayRepeatMax_8`, `scriptId_8`) VALUES (35160, 'Darnassus - Arch Druid Fandral Staghelm', 20698, 100, 1, 0, 0, 0, 0, 0, 2, 3, 0, 20702, 100, 0, 0, 0, 0, 3, 5, 25, 35, 0, 20701, 100, 0, 0, 0, 0, 8, 12, 26, 30, 0, 20699, 100, 1, 0, 0, 0, 5, 7, 17, 21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: So after comparing with the existing spell list, what you changed is you removed the main ranged spell flag from Wrath, meaning he will now do the melee attacking state animation while casting, and will not stop chasing his target, so he will be making small steps between each cast. Is that really how it works in classic? Normally ranged mobs stand still and just spam their main spell when at range. Also you removed Hurricane from the list. And you changed Rejuvenation to self only, without checking health. Does he really cast it on himself on a timer, even when on full health?

File: sql/migrations/20201010150010_world.sql [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+UPDATE `creature_template` SET `spell_list_id` = 79370 WHERE `entry` = 35160;
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: Also you are assigning High Tinker Mekkatorque's list to a non-existing creature id.

File: sql/migrations/20201010150010_world.sql [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+UPDATE `creature_template` SET `ai_name` = 'EventAI' WHERE `entry` = 7999;
+
+
+-- King Magni Bronzebeard 2784
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: He currently has 3 other spells in his c++ script that you are not using. Are you sure he does not use them in Classic?

17233 - Lay on Hands
22911 - Charge
23931 - Thunder Clap

Also you should delete the old script from core when porting it to db, and also unassign script_name from the table.

File: sql/migrations/20201010150010_world.sql [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+UPDATE `creature_template` SET `ai_name` = 'EventAI' WHERE `entry` = 7999;
```
Comment (ratkosrb) [inline sql/migrations/20201010150010_world.sql]: She is already using Event AI so this query is redundant.

File: sql/migrations/20201010150010_world.sql (hunk 1 above)
Comment (NickTyrer) [reply sql/migrations/20201010150010_world.sql]: Thanks for reviewing

File: sql/migrations/20201010150010_world.sql (hunk 2 above)
Comment (NickTyrer) [reply sql/migrations/20201010150010_world.sql]: Thanks for reviewing

**Comment by NickTyrer** [comment]
TODO: 
1. Lady Jaina Proudmoore should play sound 5882 on aggro
2. Lady Sylvanas Windrunner should play sound 5886 on aggro

File: sql/migrations/20201010150010_world.sql (hunk 7 above)
Comment (NickTyrer) [reply sql/migrations/20201010150010_world.sql]: I cant find any sources online that reflect this, according to most resources he only has the abilities in this PR

File: sql/migrations/20201010150010_world.sql (hunk 5 above)
Comment (NickTyrer) [reply sql/migrations/20201010150010_world.sql]: I cannot find any evidence that he uses Hurricane.

**Pull request merged**